File: src/main/java/toker/warbandscripts/panel/repository/ProfessionAssignmentRepository.java
package toker.warbandscripts.panel.repository;

import org.springframework.data.jpa.repository.Modifying;
import org.springframework.transaction.annotation.Transactional;
import toker.warbandscripts.panel.entity.ProfessionAssignment;

import java.util.List;

public interface ProfessionAssignmentRepository extends
        BaseRepository<ProfessionAssignment, Integer> {

    List<ProfessionAssignment> findAllByPlayerId(Integer playerId);

    @Modifying
    @Transactional
    void deleteByPlayerIdAndProfessionId(Integer playerId, Integer professionId);
}

File: src/main/java/toker/panel/repository/ProfessionAssignmentRepository.kt
package toker.panel.repository

import org.springframework.data.jpa.repository.Modifying
import org.springframework.transaction.annotation.Transactional
import toker.panel.entity.ProfessionAssignment

interface ProfessionAssignmentRepository : BaseRepository<ProfessionAssignment, Int> {
    fun findAllByPlayerId(playerId: Int): List<ProfessionAssignment>

    @Modifying
    @Transactional
    fun deleteByPlayerIdAndProfessionId(playerId: Int, professionId: Int)
}